lib/core/meta.js

```diff
@@ -71,26 +72,33 @@ export default class MetaRenderer {
 
     const clientManifest = this.renderer.resources.clientManifest
 
-    const shouldPreload = this.options.render.bundleRenderer.shouldPreload || (() => true)
-    const shouldPrefetch = this.options.render.bundleRenderer.shouldPrefetch || (() => true)
+    const shouldPreload = this.options.render.bundleRenderer.shouldPreload
+    const shouldPrefetch = this.options.render.bundleRenderer.shouldPrefetch
 
     if (this.options.render.resourceHints && clientManifest) {
       const publicPath = clientManifest.publicPath || '/_nuxt/'
 
       // Preload initial resources
       if (Array.isArray(clientManifest.initial)) {
         meta.resourceHints += clientManifest.initial
-          .filter(file => shouldPreload(file))
-          .map(
-            r => `<link rel="preload" href="${publicPath}${r}" as="script" />`
-          )
+          .map(MetaRenderer.normalizeFile)
+          .filter(({fileWithoutQuery, asType}) => shouldPreload(fileWithoutQuery, asType))
+          .map(({ file, extension, fileWithoutQuery, asType }) => {
+            let extra = ''
+            if (asType === 'font') {
```

Comment:
> I think this could be a one-liner as well:

Just thought I'd say, I'm rarely concerned with one-liners in JavaScript. I think they harm more than help, in making [code readable by humans and only incidentally executable by computers](https://www.quora.com/How-true-is-Programs-are-meant-to-be-read-by-humans-and-only-incidentally-for-computers-to-execute) :) 

So I personally would prefer the original version written by @clarkdo and would say this should be what we aim for in writing Nuxt code.

Reply:
The codes are from Vue source, as @galvez said, it maybe more readable, wont affect performance and also will be compressed by building tools.

Reply:
Well, I thought the one-liner would actually be easier to read :see_no_evil: 

@galvez I prefer to write clean code, meaning human-readable and easy to maintain. Sometimes one-liners (esp. ternaries) can help there. :relaxed:

Reply:
@clarkdo @manniL I myself tend to avoid ternaries as much as I can. I just think they look terrible, so I restrain myself to simple template var calculations :)

Looking very much forward to [do expressions](https://github.com/tc39/proposal-do-expressions) being accepted. I'm already using them in a few projects through a Babel plugin and it makes code so much easier to read! Love it :)

Reply:
```
const extra = do {
  if (asType === 'font') {
    ` type="font/${extension}" crossorigin`
  } else {
    ''
   }
}
```

Reply:
Yeah, opinions tend to diverge on this topic :see_no_evil: Some colleagues dislike ternaries as well, but I got quite used to them and now see the evaluation at the first glimpse (if they are short enough).

Do expressions look promising for several situations but I'd likely use a ternary instead of 7 lines of code.

Looking forward to pipes, call/bind operator, and partial application :pray: 

And maybe tail call optimization if it'll ever ever get implemented :see_no_evil: